### PR TITLE
DFPL-620: Allow solicitors to add colleagues to receive notifications

### PR DESCRIPTION
**Pull request by DanCatchpole**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DFPL-620

### Change description ###
 - Add new event: "Add colleagues to be notified" (only accessible to `[SOLICITORA-J]`, `[CHILDSOLICITORA-O]`)
 - This will allow a Respondent Solicitor or Child Solicitor to add/update their colleagues who will receive the same notifications they will receive
 - Add functions to UserService to map the logged in user's case roles to the person they are supposed to be representing so colleagues can be stored on the correct Solicitor.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ColleaguesToNotifyController.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            .collect(Collectors.toList());
+
+        List<String> errors;
+        if (represented.isEmpty()) {
+            errors = List.of("There is no one this user is representing on this case.");
+            return respond(caseDetails, errors);
+        }
+
+        // Use the first party we are representing - solicitors shouldn't be representing multiple
+        // todo - what about children's solicitors...
```
Comment (so99y) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ColleaguesToNotifyController.java:56]: is the `todo` for the future ticket?

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ColleaguesToNotifyController.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+        List<SolicitorRole> roles = caseRoleLookupService.getCaseSolicitorRolesForCurrentUser(caseData.getId());
+        List<WithSolicitor> represented = roles.stream()
+            .map(role -> role.getRepresentedPerson(caseData))
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .map(Element::getValue)
+            .collect(Collectors.toList());
+
+        List<String> errors;
```
Comment (so99y) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ColleaguesToNotifyController.java:49]: could be put insid the `if` block?

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ColleaguesToNotifyController.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
```
Comment (so99y) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ColleaguesToNotifyController.java:77]: could be put inside the if block?

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ColleaguesToNotifyController.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        if (represented.isEmpty()) {
+            errors = List.of("There is no one this user is representing on this case.");
+            return respond(caseDetails, errors);
+        }
+
+        if (roles.get(0).getRepresenting().equals(SolicitorRole.Representing.RESPONDENT)) {
```
Comment (so99y) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ColleaguesToNotifyController.java:83]: use variable for `roles.get(0).getRepresenting()`

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ColleaguesToNotifyController.java [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+        return respond(caseDetails);
+    }
+
+    @PostMapping("/submitted")
+    public void handleSubmitted(@RequestBody CallbackRequest callbackRequest) {
+        final CaseData caseData = getCaseData(callbackRequest);
+
+        // Send emails!
+    }
```
Comment (so99y) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ColleaguesToNotifyController.java:118]: Waiting for confirmation for implementing something?

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ColleaguesToNotifyController.java (hunk 1 above)
Comment (DanCatchpole) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ColleaguesToNotifyController.java:56]: oops left that comment in by accident - turns out childrens solicitors will be fine as even if we take the first child they represent we then copy the colleagues across all children to keep them upto date!

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ColleaguesToNotifyController.java (hunk 5 above)
Comment (DanCatchpole) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ColleaguesToNotifyController.java:118]: ah yes I think originally I was thinking we should be notifying people they were added as colleagues but doesn't look like thats a requirement so will remove this callback

**Review by so99y — APPROVED** [review]

**Review by so99y — APPROVED** [review]

**Review by so99y — APPROVED** [review]

**Pull request merged**